lib/services/identitycenter.go

```diff
@@ -0,0 +1,257 @@
+	UpdatePermissionSet(context.Context, *identitycenterv1.PermissionSet) (*identitycenterv1.PermissionSet, error)
+
+	// DeletePermissionSet deletes a specific Identity Center PermissionSet
+	DeletePermissionSet(context.Context, PermissionSetID) error
+}
+
+// IdentityCenterAccountAssignment wraps a raw identitycenterv1.AccountAssignment
+// record in a new type to allow it to implement the interfaces required for use
+// with the Unified Resource listing. IdentityCenterAccountAssignment simply
+// wraps a pointer to the underlying account record, and can be treated as a
+// reference-like type.
+//
+// Copies of an IdentityCenterAccountAssignment will point to the same record.
+type IdentityCenterAccountAssignment struct {
+	*identitycenterv1.AccountAssignment
+}
+
+// CloneResource creates a deep copy of the underlying account resource
+func (a IdentityCenterAccountAssignment) CloneResource() IdentityCenterAccountAssignment {
+	var expiry *timestamppb.Timestamp
+	if a.Metadata.Expires != nil {
+		expiry = &timestamppb.Timestamp{
+			Seconds: a.Metadata.Expires.Seconds,
+			Nanos:   a.Metadata.Expires.Nanos,
+		}
+	}
+
+	return IdentityCenterAccountAssignment{
+		AccountAssignment: &identitycenterv1.AccountAssignment{
+			Kind:    a.Kind,
+			SubKind: a.SubKind,
+			Version: a.Version,
+			Metadata: &headerv1.Metadata{
+				Name:        a.Metadata.Name,
+				Namespace:   a.Metadata.Namespace,
+				Description: a.Metadata.Description,
+				Labels:      maps.Clone(a.Metadata.Labels),
+				Expires:     expiry,
+				Revision:    a.Metadata.Revision,
+			},
+			Spec: &identitycenterv1.AccountAssignmentSpec{
+				Display:     a.Spec.Display,
+				AccountId:   a.Spec.AccountId,
+				AccountName: a.Spec.AccountName,
+				PermissionSet: &identitycenterv1.PermissionSetInfo{
+					Arn:  a.Spec.PermissionSet.Arn,
+					Name: a.Spec.PermissionSet.Name,
+				},
+			},
+		},
+	}
+}
+
+// IdentityCenterAccountAssignmentID is a strongly typed ID for an
+// IdentityCenterAccountAssignment
+type IdentityCenterAccountAssignmentID string
+
+// IdentityCenterAccountAssignments defines the operations to create and maintain
+// IdentityCenterAccountAssignment records in the service.
+type IdentityCenterAccountAssignments interface {
+	// ListAccountAssignments lists all IdentityCenterAccountAssignment record
+	// known to the service
+	ListAccountAssignments(context.Context, int, *pagination.PageRequestToken) ([]IdentityCenterAccountAssignment, pagination.NextPageToken, error)
+
+	// CreateAccountAssignment creates a new Account Assignment record in
+	// the service from the supplied in-memory representation. Returns the
+	// created record on success.
+	CreateAccountAssignment(context.Context, IdentityCenterAccountAssignment) (IdentityCenterAccountAssignment, error)
+
+	// GetAccountAssignment fetches a specific Account Assignment record.
+	GetAccountAssignment(context.Context, IdentityCenterAccountAssignmentID) (IdentityCenterAccountAssignment, error)
```

Comment:
I can see that. The rationale for doing this is all in the Identity Center integration code, which you can't see yet.

Rationale is that there are so many different resource types in flight in the IC integration, so I wanted to make using the wrong ID in the wrong place a compile-time error rather than a debugging session. The CRUD service seemed a reasonable boundary to pack/unpack the IDs, so that everything downstream is checked by the compiler.

But yeah, in isolation, it's very much overkill.